crates/core/src/config/port_allocation.rs

```diff
@@ -0,0 +1,47 @@
+use std::net::UdpSocket;
+
+pub(crate) fn find_available_port() -> std::io::Result<u16> {
+    // Bind to port 0 to let OS assign an available port
+    let socket = UdpSocket::bind("127.0.0.1:0")?;
+    socket.local_addr().map(|addr| addr.port())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
```

Comment:
Given the current implementation, hoy much sense do this tests have since we are relying on the OS basically

Reply:
Yes, tests are probably redundant now.